plugins/carbon_black_defense/komand_carbon_black_defense/connection/connection.py

```diff
@@ -1,35 +1,121 @@
-import komand
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import PluginException
+
 from .schema import ConnectionSchema, Input
-from komand.exceptions import ConnectionTestException
+
 
 # Custom imports below
 import requests
+from typing import Optional
+import json
 
 
-class Connection(komand.Connection):
+class Connection(insightconnect_plugin_runtime.Connection):
     def __init__(self):
         super(self.__class__, self).__init__(input=ConnectionSchema())
+        self.job_id = None
 
-    def connect(self, params):
+    def connect(self, params={}):
         self.logger.info("Connect: Connecting..")
         self.host = params.get(Input.URL)
         self.token = params.get(Input.API_KEY).get("secretKey")
+        self.org_key = params.get(Input.ORG_KEY)
         self.connector = params.get(Input.CONNECTOR)
+        self.headers = {"X-Auth-Token": f"{self.token}/{self.connector}"}
+
+    def get_job_id_for_enriched_event(self, criteria: dict, exclusions: dict, time_range: dict, window: str) -> Optional[dict]:
+        if self.org_key == "":
+            raise PluginException(cause="There's no org key input.",
+                                  assistance="Please add a valid org key to the connection.")
```

Comment:
I'm partially to blame here btw for suggesting this check...but we could also just make org_key a required parameter of the connection object and then we don't need this chunk of code. Its not something we have to fix now that we already wrote it but food for thought.

Reply:
It looks like the org key is a required parameter. So I don't think this code could run. Since you've already written it though, I'd just leave it in.

Reply:
pinged on slack pj for more feedback on this comment